biz.aQute.resolve/src/biz/aQute/resolve/RunResolution.java

```diff
@@ -354,4 +387,119 @@ public String report(boolean optionals) {
 	public Map<Resource, List<Wire>> getRequired() {
 		return required;
 	}
+
+	public static class CacheDTO {
+		public String					checksum;
+		public List<VersionedClause>	runbundles;
+	}
+
+	/**
+	 * Cache this resolution for the given project. A cache can be retrieved
+	 * with getCached(Project project). This requires a real resolution
+	 */
+
+	public void cache() {
+
+		assert isOK() : "can only be called for a real resolution";
+
+		try {
+			CacheDTO dto = new CacheDTO();
+			dto.checksum = project.getChecksum();
+			dto.runbundles = getRunBundles();
+
+			IO.store(ff -> {
+				JSON_CODEC.enc()
+					.to(ff)
+					.put(dto)
+					.close();
+			}, getCacheFile(project));
+		} catch (Exception e) {
+			throw Exceptions.duck(e);
+		}
+	}
+
+	/**
+	 * Get the run bundles from the cache and resolve if necessary. The run
+	 * bundles
+	 *
+	 * @param project the project to get the run bundles from
+	 * @param resolveIfNecessary if true, call resolve
+	 * @return a result, where the value can be null
+	 */
+
+	public static Result<String, String> getRunBundles(Project project, boolean resolveIfNecessary) {
```

Comment:
Why don't we use Promise to hold result/failure information here? Rather than adding a new, similar package?

Reply:
Because this one is made for the 98% when you're not doing parallel stuff. The problem is really quite different. I discussed this with Albert & Ray when you were away. But we could discuss this. I think you should give it a chance but I do not see how Promise would fit here nicely.

Reply:
Yes, I must have been out for that conversation. Is there some reason we are making a copy of the code instead of using the original artifact? https://search.maven.org/artifact/com.github.leopard2a5/result-flow/1.0.0/jar

Making a copy and shipping it in bndlib seems a wrong. Now we have to maintain the source for fixes instead of just using the original.

Reply:
I'm happy to maintain the copy like we copied other stuff. It is tiny.